pkg/azuredisk/controllerserver.go

```diff
@@ -543,31 +543,62 @@ func (d *Driver) ListSnapshots(ctx context.Context, req *csi.ListSnapshotsReques
 	if err != nil {
 		return nil, status.Error(codes.Internal, fmt.Sprintf("Unknown list snapshot error: %v", err))
 	}
+
+	return d.getEntriesAndNextToken(req, snapshotListPage)
+}
+
+// There are 4 scenarios for listing snapshots.
+// 1. StartingToken is null, and MaxEntries is null. Return all snapshots from zero.
+// 2. StartingToken is null, and MaxEntries is not null. Return `MaxEntries` snapshots from zero.
+// 3. StartingToken is not null, and MaxEntries is null. Return all snapshots from `StartingToken`.
+// 4. StartingToken is not null, and MaxEntries is not null. Return `MaxEntries` snapshots from `StartingToken`.
+func (d *Driver) getEntriesAndNextToken(req *csi.ListSnapshotsRequest, snapshotListPage compute.SnapshotListPage) (*csi.ListSnapshotsResponse, error) {
```

Comment:
could you add some unit test for this func?

Reply:
done